lib/rspec/core/memoized_helpers.rb

```diff
@@ -1,4 +1,4 @@
-require 'rspec/support/reentrant_mutex'
+RSpec::Support.require_rspec_support 'reentrant_mutex'
```

Comment:
Ah that's how it's happening, it's probably using the `require_relative` in the optimised one and that counts as a different require to the bare one.